packages/repository/src/__tests__/acceptance/has-one.relation.acceptance.ts

```diff
@@ -115,6 +115,70 @@ describe('hasOne relation', () => {
     ).to.be.rejectedWith(EntityNotFoundError);
   });
 
+  it('can PATCH hasOne instances', async () => {
+    existingCustomerId = 11;
+    const address = await controller.createCustomerAddress(existingCustomerId, {
+      street: '1 Amedee Bonnet',
+      zipcode: '69740',
+      city: 'Genas',
+      province: 'Rhone',
+    });
+
+    const persisted = await addressRepo.findById(address.zipcode);
+    expect(persisted.toObject()).to.deepEqual(address.toObject());
+
+    const patchObject = {city: 'Lyon-Genas'};
+    const arePached = await controller.patchCustomerAddress(
+      existingCustomerId,
+      patchObject,
+    );
+
+    expect(arePached.count).to.equal(1);
+    const patchedData = await addressRepo.findById(address.zipcode);
+    expect(patchedData.city).to.containEql(address.city);
+  });
+
+  it('can PUT hasOne instances', async () => {
+    existingCustomerId = 11;
+    const address = await controller.createCustomerAddress(existingCustomerId, {
+      street: '1 Amedee Bonnet',
+      zipcode: '69740',
+      city: 'Genas',
+      province: 'Rhone',
+    });
+
+    const persisted = await addressRepo.findById(address.zipcode);
+    expect(persisted.toObject()).to.deepEqual(address.toObject());
+
+    const putObject = {
+      city: 'Lyon-Genas',
+      province: 'Rhone-Alpe',
+      zipcode: '69740',
+    };
+    await controller.putCustomerAddress(existingCustomerId, putObject);
+    const putData = await addressRepo.findById(address.zipcode);
+    expect(putData.city).to.containEql(address.city);
+    expect(putData.province).to.containEql(address.province);
```

Comment:
I think we should also verify `zipcode` property and probably `customerId` too. 

In general, I don't see much value in the `PUT` (`replace`) endpoint. Because the endpoint is performing a full replace, it requires all data including the instance id (`zipcode` here). Once the client has this data, it can call regular (non-relational) endpoint instead.

I am proposing to remove `PUT` endpoint from this pull request. We can discuss later what are the use cases for `PUT` and what is the best way how to support them.

Reply:
Ok, no problem. I removed the PUT from the acceptance test.